tests/common/test_cgroups.py

```diff
@@ -34,262 +31,131 @@ def consume_cpu_time():
 
+        test_cgroup = CGroup.create(os.path.join(data_dir, "cgroups", "memory_mount"), "memory", "test_extension")
+        self.assertEqual(False, test_cgroup.is_active())
 
-def i_am_root():
-    return os.geteuid() == 0
+        with open(os.path.join(data_dir, "cgroups", "memory_mount", "tasks"), mode="wb") as tasks:
+            tasks.write(str(1000).encode())
+
+        self.assertEqual(True, test_cgroup.is_active())
+
+    def test_is_active_incorrect_file(self):
+        test_cgroup = CGroup.create(os.path.join(data_dir, "cgroups", "not_cpu_mount"), "cpu", "test_extension")
+        self.assertEqual(False, test_cgroup.is_active())
+
+        test_cgroup = CGroup.create(os.path.join(data_dir, "cgroups", "not_memory_mount"), "memory", "test_extension")
+        self.assertEqual(False, test_cgroup.is_active())
+
+
+class TestCpuCgroup(AgentTestCase):
+    def setUp(self):
+        AgentTestCase.setUp(self)
+
+    @patch("azurelinuxagent.common.osutil.default.DefaultOSUtil._get_proc_stat")
+    def test_cpu_cgroup_create(self, patch_get_proc_stat):
+        patch_get_proc_stat.return_value = fileutil.read_file(os.path.join(data_dir, "cgroups", "dummy_proc_stat"))
+        test_cpu_cg = CpuCgroup("test_extension", "dummy_path")
+
+        self.assertEqual(398488, test_cpu_cg._current_system_cpu)
+        self.assertEqual(0, test_cpu_cg._current_cpu_total)
+        self.assertEqual(0, test_cpu_cg._previous_cpu_total)
+        self.assertEqual(0, test_cpu_cg._previous_system_cpu)
+
+        self.assertEqual("cpu", test_cpu_cg.controller)
+
+    @patch("azurelinuxagent.common.osutil.default.DefaultOSUtil.get_processor_cores", return_value=1)
+    @patch("azurelinuxagent.common.osutil.default.DefaultOSUtil._get_proc_stat")
+    def test_collect(self, patch_get_proc_stat, *args):
+        patch_get_proc_stat.return_value = fileutil.read_file(os.path.join(data_dir, "cgroups", "dummy_proc_stat"))
+        test_cpu_cg = CpuCgroup("test_extension", os.path.join(data_dir, "cgroups", "cpu_mount"))
+
+        # Mocking CPU consumption
+        patch_get_proc_stat.return_value = fileutil.read_file(os.path.join(data_dir, "cgroups",
+                                                                           "dummy_proc_stat_updated"))
+
+        collected_metric = test_cpu_cg.collect()[0]
+
+        self.assertEqual("cpu", collected_metric.controller)
+        self.assertEqual("% Processor Time", collected_metric.metric_name)
+        self.assertEqual(5.114, collected_metric.value)
+
+    def test_get_current_cpu_total_exception_handling(self):
+        test_cpu_cg = CpuCgroup("test_extension", "dummy_path")
+        self.assertRaises(IOError, test_cpu_cg._get_current_cpu_total)
+
+        # Trying to raise ERRNO 20.
+        test_cpu_cg = CpuCgroup("test_extension", os.path.join(data_dir, "cgroups", "cpu_mount", "cpuacct.stat"))
+        self.assertRaises(CGroupsException, test_cpu_cg._get_current_cpu_total)
```

Comment:
Why does this raise a different exception if in both cases the file doesn't exist? Does it have something to do with cgroup file permissions? Asking because that might change when we figure out how to enable cgroups environment in Travis.